### PR TITLE
bump go dependencies

### PR DESCRIPTION
**Pull request by jkhelil**
# Changes
Bump openshift/api and openshift/client-go to provide fake security clientset

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes
None

### CODE REVIEW

**Comment by savitaashture** [comment]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/tektoncd/operator
 
-go 1.21
+go 1.22.0
```
Comment (vdemeester) [inline go.mod:3]: Any reason for `1.22.0` and not `1.22` ?

**Review by vdemeester — APPROVED** [review]
One nit, otherwise LGTM 🐯

File: go.mod (hunk 1 above)
Comment (jkhelil) [reply go.mod]: it is automatically added by go mod tidy and toolchain go1.22.4. but removing it now

Comment (vdemeester) [reply go.mod]: https://github.com/tektoncd/plumbing/pull/2043 should also help not having to remove `toolchain` in the future 🤞🏼

**Review by jkandasa — APPROVED** [review]
/lgtm

**Pull request merged**